File: appium/main_test.dart
import 'package:appium_driver/async_io.dart';
import 'package:appium_driver/src/common/flutter_finder.dart';

import 'package:test/test.dart';
import 'dart:developer';
import 'dart:io';

void main() {
  late AppiumWebDriver driver;

  setUpAll(() async {
      driver = await createDriver(
        uri: Uri.parse('http://127.0.0.1:4723/wd/hub/'),
        desired: {
          'platformName': 'android',
          'appium:automationName': 'uiautomator2',
          //'appium:app': 'https://github.com/getlantern/lantern-binaries/raw/main/lantern-installer-beta.apk?raw=true',
          'app': File('lantern-installer-arm32-debug.apk').absolute.path,
          //'appium:appPackage': 'org.getlantern.lantern',
          'appium:appActivity': 'MainActivity',
          //'appium:noReset': true,  // to ensure the device under test clean the app under test.
        });
  });

  tearDownAll(() async {
    await driver.quit();
  });

  test('turn VPN on', () async {
    var el = await driver.findElement(AppiumBy.accessibilityId('vpnSwitch'));
    el.click();
  });
}

File: appium/main_test.py
import os
import time
import unittest
from appium import webdriver
from appium.options.android import UiAutomator2Options
from appium.webdriver.common.appiumby import AppiumBy


options = UiAutomator2Options()
options.platformVersion = '10'
options.appPackage = 'org.getlantern.lantern'
options.appActivity = '.MainActivity'


appium_server_url = 'http://localhost:4723/wd/hub'

class ConnectToDevice:
    @staticmethod
    def connect_to_device(deviceName, apk_file,appPackage, appActivity ):
        capabilities = {}
        capabilities['platformName'] = 'Android'
        capabilities['deviceName'] = deviceName


        if apk_file:
            capabilities['app'] = os.path.abspath(apk_file)
        capabilities["appPackage"] = appPackage
        if appActivity:
            capabilities['appActivity'] = appActivity

        driver = webdriver.Remote('http://localhost:4723/wd/hub', capabilities)
        return driver


class TestMain(unittest.TestCase):
    driver = ConnectToDevice.connect_to_device("test-device", "../lantern-installer-all-debug.apk", 
        "org.getlantern.lantern", None)
    time.sleep(5)
    # element = driver.find_element(AppiumBy.ID,"org.getlantern.lantern:id/vpnSwitch")
    # element.click()

if __name__ == '__main__':
    unittest.main()